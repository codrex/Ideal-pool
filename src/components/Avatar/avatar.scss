.avatar {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  &__img {
    height: 3rem;
    width: 3rem;
    & img {
      width: 100%;
      min-width: 100%;
      object-fit: contain;
      object-position: center;
    }
  }
  &__text {
    color: #fff;
    font-size: 1rem;
    margin-top: 1rem;
    text-transform: capitalize;
    line-height: 18px;
  }
}
