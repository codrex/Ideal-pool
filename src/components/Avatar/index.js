// @flow
import React, { PureComponent } from 'react';
import type { Element } from 'react';
import './avatar.scss';

type Props = {
  className?: string,
  children: Element<'img'>,
  text?: string,
};
class Avatar extends PureComponent<Props> {
  static defaultProps = {
    className: '',
    text: '',
  };

  render() {
    const { className = '', children, text } = this.props;
    return (
      <div className={`avatar ${className}`}>
        <div className="avatar__img">{children}</div>
        <span className="avatar__text">{text}</span>
      </div>
    );
  }
}

export default Avatar;
