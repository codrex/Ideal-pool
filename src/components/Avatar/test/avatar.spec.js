// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Avatar from '..';

describe('Avatar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Avatar>
        <img src="" alt="" />
      </Avatar>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
