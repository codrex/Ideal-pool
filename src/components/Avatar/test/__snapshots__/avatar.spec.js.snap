// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<div
  className="'avatar' "
>
  <div
    className="avatar__img"
  >
    <img
      alt=""
      src=""
    />
  </div>
  <span
    className="avatar__text"
  />
</div>
`;
